src/Xamarin.Android.Build.Tasks/Tasks/CheckClientHandlerType.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.IO;
+using Microsoft.Android.Build.Tasks;
+using Java.Interop.Tools.Cecil;
+using Mono.Cecil;
+
+namespace Xamarin.Android.Tasks
+{
+	public class CheckClientHandlerType : AndroidTask
```

Comment:
My concern with a task such as this is that it adds yet another place where we load and process "lots" of assemblies.

Would this check be better done in an existing task such as ($DEITY forbid) `<GenerateJavaStubs/>` or `<LinkAssemblies/>` or `<LinkAssembliesNoShrink/>`?  *Or*, find a way to *share* the `DirectoryAssemblyResolver` instance *across* these tasks so we reduce how often we need to load them?

Reply:
I was trying to avoid adding this to `GenerateJavaStubs`. However we do not process `lots` of assemblies, we find the one we want and then walk up the dependency tree. 
Putting it in the Linker Step means it will only show up when packaging... which is not ideal.

Reply:
Could we narrow the scope of this one? And just error if the value is `System.Net.Http.HttpMessageHandler`?

I think that is the specific setting that will crash your app in .NET 6+.

Reply:
@jonathanpeppers [asked](https://github.com/xamarin/xamarin-android/pull/7448#discussion_r1001101602):

> Could we narrow the scope of this one?

That's a performant thought.  It also won't crash anymore, as of d3e87674187fca4ce136f5c4b460c488044e1bec.

Part of the hope was that the runtime check in d3e87674187fca4ce136f5c4b460c488044e1bec wouldn't be the *only* way to tell that the property has an invalid value, but maybe I've been overthinking it, and the existing runtime check alongside a "don't use a known bad value that is likely to be present because of project upgrades" is sufficient.

Reply:
So I did some testing and this task only added 11ms to the build. This is because it selectively targets and loads a specific assembly. I'd rather not add this code to GenerateJavaStubs as it is big enough already.